lib/tests/subscribe.rb

```diff
@@ -123,3 +123,143 @@ def validate_notify xml_s, schema_file="schemas/reginfo.xsd"
 
 end
 
+# Test that subscriptions are actively timed out on expiry
+TestDefinition.new("SUBSCRIBE - SUB timeout test") do |t|
+  ep1 = t.add_endpoint
+
+  t.add_quaff_setup do
+    ep1.register
+  end
+
+  t.add_quaff_scenario do
+    call = ep1.outgoing_call(ep1.uri)
+
+    call.send_request("SUBSCRIBE", "", {"Event" => "reg"})
+
+    # 200 and NOTIFY can come in any order, so expect either of them, twice
+    notify1 = call.recv_200_and_notify
+    call.send_response("200", "OK")
+
+    # Set the subscription to expire shortly, sleep until it is nearly expired, then expect a NOTIFY
+    call.send_request("SUBSCRIBE", "", {"Event" => "reg", "From" => notify1.headers['To'], "To" => notify1.headers['From'], "Expires" => 3})
+
+    notify2 = call.recv_200_and_notify
+    call.send_response("200", "OK")
+
+    sleep 2.5
+    notify3 = call.recv_request("NOTIFY")
+    call.send_response("200", "OK")
+
+    call.end_call
+
+    # Validate NOTIFYs are correctly formed
+    fail "NOTIFY responses have the same CSeq!" if notify1.header('CSeq') == notify2.header('CSeq')
```

Comment:
Can you check that notify2's cseq is larger than notify1, not just that it's different (and the same for the other tests where you compare NOTIFYs)